fast64_internal/oot/oot_scene_room.py

```diff
@@ -8,596 +8,626 @@
 
-	roomIndex : bpy.props.IntProperty(name = 'Room Index', default = 0, min = 0)
-	roomBehaviour : bpy.props.EnumProperty(items = ootEnumRoomBehaviour, default = "0x00")
-	roomBehaviourCustom : bpy.props.StringProperty(default = "0x00")
-	disableWarpSongs : bpy.props.BoolProperty(name = "Disable Warp Songs")
-	showInvisibleActors : bpy.props.BoolProperty(name = "Show Invisible Actors")
-	linkIdleMode : bpy.props.EnumProperty(name = "Link Idle Mode",items = ootEnumLinkIdle, default = "0x00")
-	linkIdleModeCustom : bpy.props.StringProperty(name = "Link Idle Mode Custom", default = '0x00')
+    useCustomBehaviourX: bpy.props.BoolProperty(name="Use Custom Behaviour X")
+    useCustomBehaviourY: bpy.props.BoolProperty(name="Use Custom Behaviour Y")
 
-	useCustomBehaviourX : bpy.props.BoolProperty(name = "Use Custom Behaviour X")
-	useCustomBehaviourY : bpy.props.BoolProperty(name = "Use Custom Behaviour Y")
+    customBehaviourX: bpy.props.StringProperty(name="Custom Behaviour X", default="0x00")
 
-	customBehaviourX : bpy.props.StringProperty(name = 'Custom Behaviour X', default = '0x00')
+    customBehaviourY: bpy.props.StringProperty(name="Custom Behaviour Y", default="0x00")
 
-	customBehaviourY : bpy.props.StringProperty(name = 'Custom Behaviour Y', default = '0x00')
+    setWind: bpy.props.BoolProperty(name="Set Wind")
+    windVector: bpy.props.FloatVectorProperty(name="Wind Vector", size=3)
 
-	setWind : bpy.props.BoolProperty(name = "Set Wind")
-	windVector : bpy.props.FloatVectorProperty(name = "Wind Vector", size = 3)
+    leaveTimeUnchanged: bpy.props.BoolProperty(name="Leave Time Unchanged", default=True)
+    timeHours: bpy.props.IntProperty(name="Hours", default=0, min=0, max=23)  # 0xFFFE
+    timeMinutes: bpy.props.IntProperty(name="Minutes", default=0, min=0, max=59)
+    timeSpeed: bpy.props.FloatProperty(name="Time Speed", default=1, min=-13, max=13)  # 0xA
 
-	leaveTimeUnchanged : bpy.props.BoolProperty(name = "Leave Time Unchanged", default = True)
-	timeHours : bpy.props.IntProperty(name = "Hours", default = 0, min = 0, max = 23) #0xFFFE
-	timeMinutes : bpy.props.IntProperty(name = "Minutes", default = 0, min = 0, max = 59)
-	timeSpeed : bpy.props.FloatProperty(name = "Time Speed", default = 1, min = -13, max = 13) #0xA
+    disableSkybox: bpy.props.BoolProperty(name="Disable Skybox")
+    disableSunMoon: bpy.props.BoolProperty(name="Disable Sun/Moon")
 
-	disableSkybox : bpy.props.BoolProperty(name = "Disable Skybox")
-	disableSunMoon : bpy.props.BoolProperty(name = "Disable Sun/Moon")
+    echo: bpy.props.StringProperty(name="Echo", default="0x00")
 
-	echo : bpy.props.StringProperty(name = "Echo", default = '0x00')
+    objectList: bpy.props.CollectionProperty(type=OOTObjectProperty)
 
-	objectList : bpy.props.CollectionProperty(type = OOTObjectProperty)
+    roomShape: bpy.props.EnumProperty(items=ootEnumRoomShapeType, default="ROOM_SHAPE_TYPE_NORMAL")
```

Comment:
Just noting that changing the property name here does cause the previous setting to be lost, but I'm assuming everybody's using type 0 at this point so it should be fine.

Reply:
I forgot about that, though it takes 2 seconds anyway to change that, it's not like actors where it's really annoying, you won't have a lot of rooms in your scene anyway

Reply:
And everyone really must be using type 0 ("normal") anyway at the moment, since type 1 is unsupported and type 2 was just documented